### ISSUE TITLE
[Practice Session 4] p.58 axios.put에 관해

### ISSUE THREAD

**Issue by bzs5919** [swsnu/swppfall2020#64]
58페이지에서 보면 axios.put의 인자에 url 외에 다른 값이 없습니다. 다른 예제를 보니 put에 보내는 두번째 인자로 url상의 data를 수정하는 것 같은데, 어떻게 두번째 인자를 입력하지 않고 서버의 값을 바꿀 수 있는건지 이해가 가지 않습니다.
아마 제가 뭔가를 잘못 이해하고 있어서 이런 질문을 올리고 있는 것 같은데, 혹시 질문의 의도가 파악이 안되신다면,
현재 코드에서 서버의 done 값이 수정되는 과정이 어떻게 이루어지는지 설명해주시면 정말 감사하겠습니다.

**Comment by Algy** [comment]
Django수업을 아직 들어가지 않았지만, PUT은 다음과 같은 소스로 구성되어있습니다.
```
 53     elif request.method == 'PUT':
 54         if id is None:
 55             return HttpResponseBadRequest('TodoID is not specified.')
 56         try:
 57             todo = Todo.objects.get(id=id)
 58             todo.done = not todo.done
 59             todo.save()
 60             return HttpResponse(status=204)
 61         except KeyError as e:
 62             return HttpResponseBadRequest('TodoID does not exist: {}'.format(id))
```
즉, id에 해당하는 db record를 불러와서 단순히 done을 토글만 해서 저장한 뒤, 그것을 출력하는 형태입니다. 그렇기 때문에 path상에 존재하는 id(e.g. delete `/api/todos/4/`) 외에 필요한 정보가 없어서 두번째 인자, body를 보내지 않습니다.

**Comment by bzs5919** [comment]
다시 말하면, axios.put뿐만 아니라 get delete 등의 함수들이 불렸을 때 어떤 처리를 할 지가 backend 서버에 짜여진 코드에 따라 다르다는 것인가요?

**Comment by Algy** [comment]
네 맞습니다. 해당 내용 자세한 것은 django 수업 세션때 배울 예정입니다. 미리 설명 드리자면 GET POST PUT DELETE의 대략적이고 추상적인 의미(가져오기, 생성, 수정, 삭제)는 지키긴 해야하지만 그 구체적인 내용은 backend의 구현에 달려있습니다. 가령, PUT을 done toggle으로 정하는 것은 backend 코더의 몫입니다.

**Comment by bzs5919** [comment]
넵 답변 감사합니다!
